charts/sonarqube/templates/deployment.yaml

```diff
@@ -15,7 +15,11 @@ metadata:
     app.kubernetes.io/component: {{ template "sonarqube.fullname" . }}
     app.kubernetes.io/version: {{ tpl .Values.image.tag . | quote }}
 spec:
-  replicas: {{ .Values.replicaCount }}
+  {{- if or (eq .Values.replicaCount 1.0) (eq .Values.replicaCount 0.0)}}
```

Comment:
Is there a specific limitation in helm that prevents us to use integer instead of double here?

Reply:
the `replicaCount` value is recognized as float, therefore I was setting 1.0 in the comparison. I can typecast the value and have something like `(eq (int .Values.replicaCount) 1) ` in the condition. Would that be acceptable for you? :)

Reply:
We discussed this privately and decided to remove such a "behind-the-scene" reset of the `replicaCount` value, introducing input validation for `replicaCount` (credits to [this blog post](https://austindewey.com/2020/06/13/helm-tricks-input-validation-with-values-schema-json/#:~:text=The%20values.,annotate%20and%20validate%20JSON%20documents%E2%80%9D) by Austin Dewey). The solution will be detailed in the next commit.